python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py

```diff
@@ -1,26 +1,27 @@
 from dagster import Field, StringSource, resource
 from dagster._utils.merger import merge_dicts
 
+from python_modules.dagster.dagster._config import BoolSource, IntSource
 from .file_manager import S3FileManager
 from .utils import construct_s3_client
 
 S3_SESSION_CONFIG = {
     "use_unsigned_session": Field(
-        bool,
+        BoolSource,
```

Comment:
i dont think you'd need to use `*Source` for it because they won't be secrets nor set in env vars